.github/workflows/build-test-publish.yml

```diff
@@ -51,10 +33,29 @@ jobs:
             ${{runner.os}}-
       - name: Upgrade pip
         run: |-
-          python -m pip install -U pip
+          python -m pip install -U pip versioneer
       - name: Install test dependencies
         run: |-
           python -m pip install pytest pytest-cov '.[test]'
       - name: Run unit tests
         run: |-
           python -m pytest --cov-report=term --cov=capellambse --rootdir=.
+
+  publish:
+    runs-on: ubuntu-latest
+    needs: test
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.8"
+      - name: Build packages
+        run: |-
+          echo "Building '$GITHUB_REF'"
+          python -m pip install -U pip build twine
+          python -m build
+          python -m twine check dist/*
```

Comment:
These lines represent a functional unit, therefore I put them together. Splitting everything out defeats the purpose of having build steps.

Reply:
> These lines represent a functional unit, therefore I put them together.

That might make sense if we were to maintain these lines together as a Github Action repo. But currently we don't.

Regardless, my main argument was "making the log output more manageable". Just click on the "Build packages" step in [the latest pipeline][1], and see for yourself. (It's even worse with [the version which I originally commented on][2].

[1]: https://github.com/DSD-DBS/py-capellambse/runs/4151304118?check_suite_focus=true
[2]: https://github.com/DSD-DBS/py-capellambse/runs/4139583446?check_suite_focus=true

There are two lines saying "Checking <somefile>: PASSED" at the end of the step's output log, which I presume to be from twine. I can only presume, because Github does not show any marker when the next script line starts. And even if it did, you'd have to scroll down several screenful before seeing it.

If on the other hand twine were its own pipeline step, it would get a nice little section for itself, and you could easily see whether the actual build failed or whether twine found issues after the build. And in the latter case, you wouldn't have to scroll so far to see what the actual issues were.

(And if these lines were maintained as a full-blown Github Action, the entire problem would go away, because actions can subdivide the step output into multiple smaller sections. But I still think that would be overkill for the issue at hand.)

> Splitting everything out defeats the purpose of having build steps.

How so? Isn't a build step just a small piece of the larger build run (with a label so you know what happens here at a glance)? I don't see a problem with this. (Again, I *would* see a problem with splitting them up into multiple Github Actions, but these aren't Actions, these are just simple lines of bash script.)

Reply:
It would've been nicer if GH Actions would collect the output per command. I'll split it up.